sympy/solvers/solveset.py

```diff
@@ -521,6 +521,15 @@ def _solve_as_rational(f, symbol, domain):
         invalid_solns = _solveset(h, symbol, domain)
         return valid_solns - invalid_solns
 
+def _solve_hyperbolic(f, symbol, domain):
+
+    # transform the hyperbolic functions into non hyperbolic
+    # redo the calulcations from the beginning
+
+    f = trigsimp(f)
+    f = f.rewrite(exp)
+    return solveset(f,symbol,domain)
```

Comment:
I think that maybe this should use `_solveset` rather than `solveset`. Also there should be spaces after the commas.